### PR TITLE
Feat(camelot): add swap support

### PR DESCRIPTION
**Pull request by Quazia**

### CODE REVIEW

File: packages/camelot/src/Camelot.test.ts [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    })
+    test('should pass filter with valid ETH transactions', async () => {
+      const transaction = SWAP_ETH
+      const filter = await swap({
+        chainId: ARBITRUM_CHAIN_ID,
+        contractAddress: CAMELOT_ROUTER,
+        tokenOut: '0xBfbCFe8873fE28Dfa25f1099282b088D52bbAD9C',
+        amountOut: GreaterThanOrEqual(parseEther('0.0005')),
+      })
+      console.log(filter)
```
Comment (jonathandiep) [inline packages/camelot/src/Camelot.test.ts:39]: Can remove this log

File: packages/camelot/src/abi.ts [hunk 2]
```diff
@@ -0,0 +1,247 @@
...
+      { internalType: 'uint256', name: 'amountAMin', type: 'uint256' },
+      { internalType: 'uint256', name: 'amountBMin', type: 'uint256' },
+      { internalType: 'address', name: 'to', type: 'address' },
+      { internalType: 'uint256', name: 'deadline', type: 'uint256' },
+      { internalType: 'bool', name: 'approveMax', type: 'bool' },
+      { internalType: 'uint8', name: 'v', type: 'uint8' },
+      { internalType: 'bytes32', name: 'r', type: 'bytes32' },
+      { internalType: 'bytes32', name: 's', type: 'bytes32' },
+    ],
+    name: 'removeLiquidityWithPermit',
```
Comment (jonathandiep) [inline packages/camelot/src/abi.ts:197]: Dumb q: If we include the other functions inside the abi, is it possible that they could trigger a successful action? Like if the params match

e.g. Can this `removeLiquidity` fx trigger a completion for a `swap` action if it had matching params?

**Review by jonathandiep — APPROVED** [review]

File: packages/camelot/src/abi.ts (hunk 2 above)
Comment (Quazia) [reply packages/camelot/src/abi.ts:197]: Yeah that's possible - I just narrowed the included ABIs that's easier than seeing if what you're describing could happen lol

File: packages/camelot/src/Camelot.test.ts (hunk 1 above)
Comment (Quazia) [reply packages/camelot/src/Camelot.test.ts:39]: I removed a few other hanging ones - I'm so bad at removing logs I'd make a terrible lumberjack

**Pull request merged**